### PR TITLE
YONK-15: API optimisations using course blocks

### PR DESCRIPTION
**Pull request by aamishbaloch**
APIs are updated and optimised using course blocks.

### CODE REVIEW

**Comment by aamishbaloch** [comment]
@ziafazal Can you review this?

**Comment by aamishbaloch** [comment]
@ziafazal Done with the refactoring you suggested. Can you review the updates?

File: edx_solutions_api_integration/courses/views.py [hunk 1]
```diff
@@ -309,47 +201,71 @@ def _manage_role(course_descriptor, user, role, action):
...
-            depth
-        )
-        data['content'] = data['children']
-        data.pop('children')
+    protocol = 'http'
+    if request.is_secure():
+        protocol = protocol + 's'
+
+    base_content_uri = '{}://{}/api/server/courses'.format(
+        protocol,
```
Comment (ziafazal) [inline edx_solutions_api_integration/courses/views.py]: Can we clear the clutter around protocol and use some like this `request.scheme`?

File: edx_solutions_api_integration/courses/views.py [hunk 2]
```diff
@@ -518,19 +438,16 @@ def get(self, request, course_id, content_id):
...
-        response_data[element_name] = _serialize_content_children(
-            request,
-            course_id,
-            children
+        usage_key = usage_key.replace(course_key=modulestore().fill_in_run(usage_key.course_key))
+        data_blocks = get_blocks(
+                    request,
+                    usage_key,
+                    depth=1,
+                    requested_fields=['children', 'display_name', 'type', 'due']
```
Comment (ziafazal) [inline edx_solutions_api_integration/courses/views.py]: Can we put requested_fields in a module level constant?

File: edx_solutions_api_integration/courses/views.py [hunk 3]
```diff
@@ -309,47 +201,71 @@ def _manage_role(course_descriptor, user, role, action):
...
-    resource_uri = '{}/overview/'.format(base_uri_without_qs)
-    data['resources'].append({'uri': resource_uri})
-    resource_uri = '{}/updates/'.format(base_uri_without_qs)
-    data['resources'].append({'uri': resource_uri})
-    resource_uri = '{}/static_tabs/'.format(base_uri_without_qs)
-    data['resources'].append({'uri': resource_uri})
-    resource_uri = '{}/users/'.format(base_uri_without_qs)
-    data['resources'].append({'uri': resource_uri})
-    return data
+        if usage_key is None:
```
Comment (ziafazal) [inline edx_solutions_api_integration/courses/views.py]: Can we add an inline comment to explain when this code?

**Pull request merged**